Harvest.Core/Services/InvoiceService.cs

```diff
@@ -232,5 +234,29 @@ public async Task<List<int>> GetCreatedInvoiceIds()
             return await _dbContext.Invoices.Where(a => a.Status == Invoice.Statuses.Created).Select(a => a.Id)
                 .ToListAsync();
         }
+
+        /// <summary>
+        /// Automatically closeout projects that are 18 days or older since closeout was initiated.
+        /// </summary>
+        /// <param name="daysBeforeAutoCloseout">Negative Value</param>
+        /// <returns></returns>
+        public async Task<int> AutoCloseoutProjects(int daysBeforeAutoCloseout)
+        {
+            var dateCutoff = DateTime.UtcNow.AddDays(daysBeforeAutoCloseout);
+
+            var count = 0;
+            var potentialProjectIds = await _dbContext.Projects.Where(a => a.IsActive && a.Status == Project.Statuses.PendingCloseoutApproval).Select(a => a.Id).ToArrayAsync();

```

Comment:
Maybe not worth the effort but i don't like doing this via the history.  I think a `LastUpdateOn` db field would make things a lot cleaner here, and then we'd just update that flag when a project changes status.

Reply:
@srkirkland I'm ok with that, but that would be a new DB field

Reply:
yea.  i think it'd be better though